### PR TITLE
asyncio.wait replaced with Semaphore, fixed compatibility issues

### LINKED ISSUE #91: Use semaphore to replace the current
https://github.com/taskiq-python/taskiq/blob/318b25b40e2cc9753c6e4c76b462a9627ecb0b62/taskiq/cli/worker/async_task_runner.py#L47-L49
We can use `asyncio.Semaphore` to replace the current scheme, and it is not recommended to set `max_async_tasks` less than 1, the infinite number of tasks will only cause unexpected situations.
Use `asyncio.Semaphore` to limit the number of tasks, assign task scheduling to the event loop, and avoid using `asyncio.wait` to increase power consumption concurrently.

### PR DESCRIPTION
**Pull request by vvanglro**
1. use semaphore, add test
2. change default max_async_tasks num
3. fix py39 has removesuffix
4. fix pre-commit

Please review it.
This PR will close #91.

### CODE REVIEW

**Comment by s3rius** [comment]
@vvanglro, can you please rebase onto develop? I updated CI so jobs must run now.

**Comment by vvanglro** [comment]
CI looks like it needs your approval.

**Comment by s3rius** [comment]
You've done a great job. But isn't 10 tasks is too low number by default? And also, now you cannot set maximum number of tasks to `-1`.

**Comment by vvanglro** [comment]
There are 10 concurrent tasks by default for each process, and 50 for 5 processes, which I think is quite a lot. Of course, users can also pass in the number by themselves.

**Comment by s3rius** [comment]
Yes, you're right. It looks like it would work well. Thank you.

**Review by s3rius — APPROVED** [review]
Looks good to me.

File: taskiq/abc/broker.py [hunk 1]
```diff
@@ -234,15 +234,22 @@ def inner(
...
+                            )
+                        else:
+                            fmodule = ".".join(
+                                sys.argv[0]
+                                .removesuffix(
+                                    ".py",
+                                )
+                                .split(
+                                    os.path.sep,
+                                ),
```
Comment (booqoffsky) [inline taskiq/abc/broker.py:251]: Hello! What do you think about not splitting the logic into two parts and using only a method that is suitable for all versions of python?

Comment (vvanglro) [reply taskiq/abc/broker.py:251]: Yes, it's great 😌

**Review by booqoffsky — APPROVED** [review]
A W E S O M E

**Pull request merged**